numpy/core/_add_newdocs_scalars.py

```diff
@@ -283,6 +283,13 @@ def add_newdoc_for_scalar_type(obj, fixed_aliases, doc):
     ``1970-01-01T00:00:00``.
     If created from string, the string can be in ISO 8601 date
     or datetime format.
+    
+    When parsing a string to create a datetime object, if the string contains
+    a trailing timezone (A 'Z' or a timezone offset), the timezone will be 
+    dropped and a User Warning is given.
+    
+    Datetime64 object should be considered to be UTC and therefore have an 
+    offset of +0000.
```

Comment:
```suggestion
    For the purpose of parsing time offsets the timezone naive `datetime64`
    behaves as if converting to UTC.
```
Maybe?  I don't think the old sentence is correct, the warning is there for a reason after all.

Probably this could be part of a `Notes` section?